server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/rfc/model/RequestObject.scala

```diff
@@ -0,0 +1,39 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.rfc.model
+
+import org.apache.james.jmap.rfc.model.Invocation.Invocation
+
+object RequestObject {
+
+  import play.api.libs.json._
+
+  case class Capability(value: String) extends AnyVal
+  implicit val capabilityFormat: Format[Capability] = Json.valueFormat[Capability]
+
+  case class RequestObject(using: Seq[Capability], methodCalls: Seq[Invocation], createdIds: Option[CreatedIds] = None)
+  implicit val requestObjectRead: Format[RequestObject] = Json.format[RequestObject]
```

Comment:
if you only need a Read, no need to generate a Format

Reply:
We need both Read and Write.